src/libmongoc/src/mongoc/mongoc-topology-private.h

```diff
@@ -191,4 +191,31 @@ _mongoc_topology_request_scan (mongoc_topology_t *topology);
 
 void
 _mongoc_topology_bypass_cooldown (mongoc_topology_t *topology);
+
+
+typedef enum {
+   MONGOC_SDAM_APP_ERROR_BEFORE_HANDSHAKE,
+   MONGOC_SDAM_APP_ERROR_AFTER_HANDSHAKE
+} _mongoc_sdam_app_error_when_t;
```

Comment:
I don't have a strong opinion here. If it seems simpler to be a boolean it shall be done. Jeremy had also asked about this, since `MONGOC_SDAM_APP_ERROR_BEFORE_HANDSHAKE` is not used outside of tests currently.